rajawali/src/main/java/c/org/rajawali3d/engine/EngineImpl.java

```diff
@@ -35,76 +33,90 @@
+    @GuardedBy("renderViews")
+    protected final List<RenderView> renderViews;
+
+    // List of all SurfaceCallbacks registered with the Engine
+    @GuardedBy("surfaceCallbacks")
+    protected final List<SurfaceCallback> surfaceCallbacks;
 
     @NonNull
-    protected Context context; // Context the renderer is running in
+    protected Context context; // Android context the engine is running in
 
     protected ISurface surface; // The rendering surface
 
-    protected int defaultViewportWidth;
-    protected int defaultViewportHeight; // The default width and height of the GL viewport
+    protected int surfaceWidth;
+    protected int surfaceHeight; // The width and height of the surface
 
     // Frame related members
     protected ScheduledExecutorService timer; // Timer used to schedule drawing
     protected double frameRate; // Target frame rate to render at
     protected int frameCount; // Used for determining FPS
     protected double lastMeasuredFPS; // Last measured FPS value
     protected OnFPSUpdateListener fpsUpdateListener; // Listener to notify of new FPS values.
-    private long startTime = System.nanoTime(); // Used for determining FPS
-    private long lastRender; // Time of last rendering. Used for animation delta time
-    private long renderStartTime;
+    private long framesStartTime; // Time of last startFrames()
+    private long fpsStartTime = System.nanoTime(); // Used for determining FPS
+    private long lastFrameTime; // Time of last frame. Used for animation delta time
 
     // In case we cannot parse the version number, assume OpenGL ES 2.0
     protected int glesMajorVersion = 2; // The GL ES major version of the surface
     protected int glesMinorVersion = 0; // The GL ES minor version of the surface
 
     /**
-     * The scene currently being displayed.
-     */
-    @GuardedBy("nextRenderableLock")
-    private Renderable currentRenderable;
-
-    private Renderable nextRenderable; //The scene the renderer should switch to on the next frame.
-    private final Object nextRenderableLock = new Object(); //Scene switching lock
-
-    /**
-     * The thread id of the GL thread for this {@link Renderer}. This should be set once and left untouched.
+     * The thread id of the GL thread for this {@link Engine}. This should be set once and left untouched.
      */
     private AtomicLong glThread;
 
-    public RendererImpl(@NonNull Context context) {
+    public EngineImpl(@NonNull Context context) {
         RajLog.i(context.getString(R.string.renderer_start_header));
         RajLog.i(context.getString(R.string.renderer_start_message));
 
         this.context = context;
         frameRate = getRefreshRate();
 
-        renderables = Collections.synchronizedList(new ArrayList<Renderable>());
+        renderModels = Collections.synchronizedList(new ArrayList<RenderModel>());
+        renderViews = Collections.synchronizedList(new ArrayList<RenderView>());
+
```

Comment:
OK on line length. My Mac screen can handle that easily too. I just tend to use a structure view on the right that cuts down on the available editor window, but I’ll adapt.